src/schedule/ll_schedule.c

```diff
@@ -353,6 +354,33 @@ static int schedule_ll_task(void *data, struct task *task, uint64_t start,
 
 	pdata->period = period;
 
+	/* for full synchronous domain, calculate ratio and initialize skip_cnt for task */
+	if (sch->domain->full_sync) {
+		pdata->ratio = 1;
+		pdata->skip_cnt = (uint16_t)(-1);
+
+		/* get the registrable task */
+		list_for_item(tlist, &sch->tasks) {
+			curr_task = container_of(tlist, struct task, list);
+			pipe_task = pipeline_task_get(curr_task);
+
+			/* registrable task found */
+			if (pipe_task->registrable) {
+				registrable_task = curr_task;
+				break;
+			}
+		}
+
+		/* we found a registrable task */
+		if (registrable_task) {
```

Comment:
Be careful that this registrable_task could be invalid if the task is scheduled earlier than the registrable_task.

Reply:
> Be careful that this registrable_task could be invalid if the task is scheduled earlier than the registrable_task.

yes, I'll fix this.